python_modules/dagster/dagster/serdes/serdes.py

```diff
@@ -103,33 +122,44 @@ def whitelist_for_serdes(__cls: Type) -> Type:
 
 @overload
 def whitelist_for_serdes(
-    __cls: None = None, *, serializer: Type["Serializer"]
+    __cls: None = None, *, serializer: Optional[Type["Serializer"]], storage_name: Optional[str]
 ) -> Callable[[Type], Type]:
     ...
 
 
 def whitelist_for_serdes(
-    __cls: Optional[Type] = None, *, serializer: Optional[Type["Serializer"]] = None
+    __cls: Optional[Type] = None,
+    *,
+    serializer: Optional[Type["Serializer"]] = None,
+    storage_name: Optional[str] = None,
```

Comment:
I see what you mean, but there are use cases for both arguments at once. Like suppose `TableRecord` (which has a custom serializer) was renamed, then you would want to pass both the custom serializer and `storage_name`.

The real assumption is that if a `NamedTupleSerializer` is passed, it's a `DefaultNamedTupleSerializer` descendant-- what if we just enforce that?

Reply:
> it's a DefaultNamedTupleSerializer descendant

and calling in to the parent impl instead of purely overriding it. Likely the case but still a bit circuitous. 

Not a blocking concern. Enforce what you think make sense for now, we can change it later if we want.

Reply:
Alternatively we could just leave out the `storage_name` from the decorator and make separate explicit calls to `register_serialized_name`/`register_deserialized_name`-- probably cleaner.

Reply:
> make separate explicit calls 

if we go that way would consider having one call that updates both sides of the bi-directional mapping